### PR TITLE
Reimplement HandleInvalidTagDeclDefinition without clang support.

### PR DESCRIPTION
**Pull request by vgvassilev**

### CODE REVIEW

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
Looks like we one of the failures is due to infrastructure and the other fails elsewhere too.

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by vgvassilev** [comment]
ping...

**Comment by Axel-Naumann** [comment]
I suppose you're pinging me here? It's of much greater success to ping me during work days :-) I'll try to review this week; the release needs to go first.

**Comment by vgvassilev** [comment]
There is not much to review here but ok.

**Review by oshadura — COMMENTED** [review]
For me, it looks like it is a missing upstreamed patch in LLVM and two reverts of existing comments...

**Review by oshadura — APPROVED** [review]
Ops, I wanted to approve instead

**Comment by vgvassilev** [comment]
Thanks! @Axel-Naumann, let us know if you have comments.

**Comment by vgvassilev** [comment]
@phsft-bot build!

**Comment by Axel-Naumann** [comment]
Looks like the review by@oshadura missed a removed code branch. Was that removed intentionally?

**Comment by vgvassilev** [comment]
The one in clang or in cling? I think they are both intentional.

**Review by Axel-Naumann — COMMENTED** [review]
Sorry, failed to actually submit the review. Here it is. And the removed code branch still exists in llvm's `main`, so I'd like to have an explanation why it was removed here.

File: interpreter/llvm/src/tools/clang/lib/Sema/SemaTemplateInstantiate.cpp [hunk 1]
```diff
@@ -2235,21 +2235,13 @@ Sema::InstantiateClass(SourceLocation PointOfInstantiation,
...
-  } else {
-    Consumer.HandleTagDeclDefinition(Instantiation);
-
-    // Always emit the vtable for an explicit instantiation definition
-    // of a polymorphic class template specialization. Otherwise, eagerly
-    // instantiate only constexpr virtual functions in preparation for their use
-    // in constant evaluation.
-    if (TSK == TSK_ExplicitInstantiationDefinition)
-      MarkVTableUsed(PointOfInstantiation, Instantiation, true);
-    else if (MightHaveConstexprVirtualFunctions)
```
[the remark this answers is not in the document]
Comment (Axel-Naumann) [reply interpreter/llvm/src/tools/clang/lib/Sema/SemaTemplateInstantiate.cpp:2249]: ping @vgvassilev this still needs to be addressed!

[the remark this answers is not in the document]
Comment (vgvassilev) [reply interpreter/llvm/src/tools/clang/lib/Sema/SemaTemplateInstantiate.cpp:2249]: Yeah, indeed that was missing. It is readded in https://github.com/root-project/root/pull/8135

Luckily, being patch-free that would have been 'transiently' fixed with the next upgrade.

**Pull request merged**